server/api.js

```diff
@@ -1,9 +1,59 @@
 import { Router } from "express";
+import pool from "./db";
 
 const router = Router();
+const bcrypt = require("bcrypt");
 
-router.get("/", (_, res) => {
-	res.json({ message: "Hello, world!" });
+router.get("/users", (req, res) => {
+	res.json("hello world");
 });
 
+router.post("/signup", async (req, res) => {
+	try {
+		//this hides the password
+		const hashedPassword = await bcrypt.hash(req.body.password, 10);
+		const newUser = {
+			firstName: req.body.firstName,
+			lastName: req.body.lastName,
+			username: req.body.username,
+			cohortId: req.body.cohortId,
+			password: hashedPassword,
+		};
+		//checking if username already exist
+		pool
+		.query("SELECT user_name FROM users WHERE user_name=$1", [newUser.username])
+		.then((result) => {
+			if (result.rows.length > 0) {
+				res.status(400).send("Username already exist!");
+			} else {
+				//Checking if cohort ID exist
+				pool.query("SELECT id FROM cohorts WHERE id=$1", [newUser.cohortId])
```

Comment:
so this here wouldn't be needed. In the frontend signup form the cohorts dropdown will be populated from a GET request from the /cohorts endpoint. Every cohort will be a valid one - for sure, that's what I tried to explain last night very badly lmao

Reply:
This is done.